File: Fourth Semester - Design and Programming Environments/Java/ProiectJavaNetworking/TravelModel/src/java/org/example/model/validator/ReservationValidator.java
package org.example.model.validator;


import org.example.model.Reservation;

public class ReservationValidator implements Validator<Reservation> {
    @Override
    public void validate(Reservation entity) {
        if (entity == null) {
            throw new IllegalArgumentException("Agency is null");
        }
        if (entity.getNumberOfReservedSeats() == null) {
            throw new ValidationException("Number of reserved places is null");
        }
        if (entity.getTour() == null) {
            throw new ValidationException("Tour is null");
        }
    }
}

File: Fourth Semester - Design and Programming Environments/C#/ProiectC#Networking/TravelModel/model/validator/ReservationValidator.cs
namespace TravelModel.model.validator;

public class ReservationValidator: IValidator<Reservation>
{
    public bool Validate(Reservation entity)
    {
        if (entity == null)
        {
            throw new ArgumentNullException(nameof(entity), "Reservation cannot be null");
        }

        if (entity.ClientName == null)
        {
            throw new ValidationException("Client name cannot be empty");
        }
        
        if (entity.ClientContact == null)
        {
            throw new ValidationException("Client contact cannot be empty");
        }
        
        if (entity.Tour == null)
        {
            throw new ValidationException("Tour cannot be empty");
        }
        
        if (entity.NumberOfReservedSeats == null)
        {
            throw new ValidationException("Number Of Reserved Seats cannot be empty");
        }
        return true;
    }
}